routes/profile.js

```diff
@@ -109,6 +110,9 @@ function profileRoutes (server, options, next) {
         })
 
         .then(function (session) {
+          if (session.account.id !== id) {
+            throw errors.accountIdConflict(session.account.id)
+          }
```

Comment:
here you should compare `session.account.id + '-profile'` to `id`. And call the `errors.accountIdConflict` with `session.account.id + '-profile'`, see my comment below